File: codegen/codegen-erlang/src/test/resources/ir/rest_json_encode_get_name_response.expected.erl
%% @doc Encode HTTP response for smithy.beam.demo.http#GetName.
-spec encode_get_name_response(get_name_output()) -> #http_response{}.
encode_get_name_response(#get_name_output{name = Name}) ->
    Headers = [{<<"Content-Type">>, <<"application/json">>}],
    BodyMap = maps:filter(
        fun(_, V) ->
            V =/= undefined
        end,
        #{<<"name">> => Name}
    ),
    Body = jsone:encode(BodyMap),
    #http_response{
        status = 200,
        headers = Headers,
        body = Body
    }.

File: codegen/codegen-elixir/src/test/resources/ir/rest_json_encode_get_name_response.expected.ex
@doc "Encode response for smithy.beam.demo.http#GetName."
@spec encode_get_name_response(HttpServiceTypes.GetNameOutput.t()) :: map()
def encode_get_name_response(output = %Types.GetNameOutput{name: name}) do
  body_map =
    %{"name" => output.name}
    |> Enum.reject(fn {_, v} -> Kernel.is_nil(v) end)
    |> Map.new()
  body = Jason.encode!(body_map)
  headers = [{"Content-Type", "application/json"}]
  %{
    status: 200,
    headers: headers,
    body: body
  }
end